config/proto/probe.proto

```diff
@@ -0,0 +1,51 @@
+syntax = "proto3";
+
+package = hermes;
+
+import "github.com/google/cloudprober/metrics/proto/dist.proto";
+
+option go_package = "github.com/googleinterns/step224-2020/config/proto"
+
+message HermesProbeDef {
+  required string probe_name= 1;
+  required targets.TargetsDef targets = 2;
+
+  enum TargetSystem {
+    GCS = 1;
+    CEPH = 2;
+  }
+
+  // Type of Probe
+  enum ProbeType {
+    WRITE = 1;
+    DELETE = 2;
+    CHECK = 3;
+  }
+
+  TargetSystem target_system = 3;
+  ProbeType probe_type = 4;
+
+  optional int32 interval_sec = 5 [default = 3600]; // Probing interval in seconds
+
+  // Probes will not retry, they will fail if they timeout.
+  optional int32 timeout_sec = 6 [default = 60]; // Timeout in seconds
+
+  // If specified, latency is stored as a distribution
+  optional metrics.Dist probe_latency_distribution = 7;
+  optional metrics.Dist api_call_latency_distribution = 8;
+
+  // Valid values: "ns", "us" (or "µs"), "ms", "s", "m", "h".
+  optional string latency_unit = 9 [default = "us"];
+
+  // Additional labels for the probe latency and API call latency distribution metrics.
+  // Add as key-value pairs
+  repeated AdditionalLabel probe_latency_additional_label = 10;
+  repeated AdditionalLabel api_call_latency_additional_label = 11;
+}
+
+// Simple Key-Value pair for additional metric labels
+message AdditionalLabel {
+  required string key = 1;
+  required string value = 2;
+}
```

Comment:
Since this is a key-value pair, you could consider using a map [1] instead of AdditionalField. Just keep in mind that maps cannot have repeated, but the map syntax should be equivalent to what your use of AdditionalLabel here.

[1] https://developers.google.com/protocol-buffers/docs/proto#maps

Reply:
We will probably have several additional labels for the distribution metrics, so we need to be able to use `repeated` here.

I might extract this into a new metric proto though and add the majority of the shared labels directly to the latency distributions.

Reply:
A map by definition can hold many key-value pairs. As I said using a single map, you achieve the same result as having repeated key-value messages.

Reply:
Yes, of course.. I think I forgot how maps work for a minute.. I think that a map would be a much cleaner solution, but using repeated AdditionalLabels should be more compatible with Cloudprober's configs, so when we implement a config that extends the Cloudprober probe config, we don't have duplicate fields (it uses AdditionalLabel too).